### PR TITLE
fix: Create separate plugin client package

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Fixes https://github.com/cloudquery/cloudquery/issues/12643

Had to refactor the templating engine code a bit as now we have different template files with the same name (`client.go.tpl` under `client` and under `resources/plugin`.

Go templating engine doesn't load templates recursively so I used `fs.WalkDir` to avoid having to manually map template files to output files.

### CODE REVIEW

File: scaffold/cmd/source.go [hunk 1]
```diff
@@ -57,44 +44,47 @@ type scaffoldData struct {
...
-	)
-	if err != nil {
-		return fmt.Errorf("failed to parse templates: %w", err)
-	}
-
 	data := scaffoldData{
 		Org:  org,
 		Name: name,
 	}
-	for templatePath, filePath := range scaffoldTemplates {
```
Comment (erezrokah) [inline scaffold/cmd/source.go:75]: Before we had an hardcoded map of template names filenames to output files. This didn't allow different templates with the same base name of a file (e.g. `client/client.go.tpl` and `plugin/client.go.tpl`)

File: scaffold/cmd/source.go [hunk 2]
```diff
@@ -57,44 +44,47 @@ type scaffoldData struct {
...
-		baseDir := path.Dir(fullPath)
-		if err := os.MkdirAll(baseDir, 0755); err != nil {
-			return fmt.Errorf("failed to create directory %s: %w", baseDir, err)
+		if d.IsDir() {
+			return nil
 		}
-		if strings.HasSuffix(filePath, ".go") {
-			formattedContent, err := format.Source(content)
+		if strings.HasSuffix(templatePath, ".tpl") {
+			tpl, err := template.New(filepath.Base(templatePath)).ParseFS(sourceFS, templatePath)
```
Comment (erezrokah) [inline scaffold/cmd/source.go:60]: Mostly the same code as before, and now getting the output path based on the template filename

File: scaffold/cmd/templates/source/client/client.go.tpl [hunk 3]
```diff
@@ -2,25 +2,13 @@ package client
 
 import (
 	"context"
-	"encoding/json"
```
Comment (erezrokah) [inline scaffold/cmd/templates/source/client/client.go.tpl:5]: Most of the code here was moved to `plugin/client.go.tpl`

File: scaffold/cmd/templates/source/resources/plugin/client.go.tpl [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	logger    zerolog.Logger
+	config    client.Spec
+	tables    schema.Tables
+	syncClient *client.Client
+	scheduler *scheduler.Scheduler
+
+	plugin.UnimplementedDestination
+}
+
+func New(ctx context.Context, logger zerolog.Logger, spec []byte, opts plugin.NewClientOptions) (plugin.Client, error) {
```
Comment (disq) [inline scaffold/cmd/templates/source/resources/plugin/client.go.tpl:28]: Maybe call this `Configure` so it doesn't get confused with `client.New`? Currently we have 13 `New*` vs. 21 `Configure` in the monorepo source plugins.

**Review by disq — APPROVED** [review]

**Pull request merged**